BUILDING.md

```diff
@@ -0,0 +1,82 @@
+components are placed directly in the bin directory while framework components end up in
+subdirectories net-2.0, net4.0, sl-5.0, portable and netcf-3.5. Future platform
+builds will cause new subdirectories to be created.
+
+## Build Script
+
+We use **Cake** (http://cake-build.net) to build NUnit for distribution. The primary script that controls
+building, running tests and packaging is build.cake. We modify build.cake when we need to add new 
+targets or change the way the build is done. Normally build.cake is not invoked directly but through
+build.ps1 (on Windows) or build.sh (on Linux). These two scripts are provided by the Cake project
+and ensure that Cake is properly installed before trying to run the cake script. This helps the
+build to work on CI servers using newly created agents to run the build and we generally run it
+the same way on our own machines.
+
+The build shell script and build.cmd script are provided as an easy way to run the above commands.
+In addition to passing their arguments through to build.cake, they can supply added arguments
+through the CAKE_ARGS environment variable. The rest of this document will assume use of these commands.
+
+There is one case in which use of the CAKE_ARGS environment variable will be essential, if not necessary.
+If you are running builds on a 32-bit Windows system, you must always supply the -Experimental argument
+to the build. Use set CAKE_ARGS=-Experimental to ensure this is always done and avoid having to type
+it out each time.
+
+Key arguments to build.cmd / build:
+ * -Target, -t <task>                 The task to run - see below.
+ * -Configuration, -c [Release|Debug] The configuration to use (default is Release)
+ * -Experimental, -e                  Use the experimental build of Roslyn
+
+The build.cake script contains a large number of interdependent tasks. The most 
+important top-level tasks to use are listed here:
+ * Build               Builds everything. This is the default if no target is given.
+ * Rebuild             Cleans the output directory and builds everything
+ * Test                Runs all tests. Dependent on Build.
+ * TestAll             Runs all tests. Dependent on Build.
+ * TestAllFrameworks   Runs all framework tests. Dependent on Build.
+ * Test45              Tests the 4.5 framework without building first.
+ * Test40              Tests the 4.0 framework without building first.
+ * Test20              Tests the 2.0 framework without building first.
+ * TestPortable        Tests the portable framework without building first.
+ * TestSL              Tests the Silverlight framework without building first.
+ * TestCF              Tests the compact framework without building first.
+ * TestEngine          Runs all engine tests. Dependent on Build.
+ * TestConsole         Runs the console tests. Dependent on Build.
+ * Package             Creates all packages without building first. See Note below.
```

Comment:
I added a fence before and after so the spacing would be respected. That seems to be the least intrusive when viewed as a text file.

I think this exercise may demonstarate that markdown is not such a great idea for files that will almost always be used locally and viewed as text. However, I don't want to go back at the moment. We can revisit the format of such files at another time.